File: Day 6/4 Reverse LinkedList/solution.cs
public ListNode ReverseList(ListNode head, int count) {
    if (head == null || head.next == null) {
        return head;
    }
    
    ListNode prev = null;
    ListNode curr = head;
     ListNode next = null;
    // ListNode next = curr.next;

    while (curr != null) {
        
        // next = curr.next;
        temp = curr.next;


        curr.next = prev;

        //prev
        prev = curr;


        curr = temp;
    }
    return prev;
}

File: Day 6/4 Reverse LinkedList/solution.js
function reverseList(head) {
    if (!head || !head.next) {
        return head;
    }
    let prev = null;
    let curr = head;
    let next = null;
    while (curr) {
        next = curr.next;
        curr.next = prev;
        prev = curr;
        curr = next;
    }
    return prev;
}
